src/main.rs

```diff
@@ -73,12 +73,20 @@ fn run() -> Result<(), MainError> {
         eprintln!("> {command}\n");
         run_script(name, command, cwd)
     };
+    let command_string = |name: &str, args: Vec<String>| {
+        let mut commands = Vec::with_capacity(args.len() + 1);
+        commands.push(name.to_string());
+        let mut args = args;
+        commands.append(&mut args);
+        commands.join(" ")
+    };
     match cli.command {
         cli::Command::Run(args) => {
             let (cwd, manifest) = cwd_and_manifest()?;
             if let Some(name) = args.script {
                 if let Some(command) = manifest.scripts.get(&name) {
-                    print_and_run_script(&manifest, &name, command, &cwd)
+                    let commands = command_string(command, args.args);
+                    print_and_run_script(&manifest, &name, &commands, &cwd)
```

Comment:
Why plural?

```suggestion
                    let command = command_string(command, args.args);
                    print_and_run_script(&manifest, &name, &command, &cwd)
```

You don't need to be concerned about name collision, Rust will shadow the name.